compiler/rustc_codegen_ssa/src/mir/debuginfo.rs

```diff
@@ -222,7 +222,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         }
     }
 
-    fn dbg_loc(&self, source_info: mir::SourceInfo) -> Option<Bx::DILocation> {
+    pub fn dbg_loc(&self, source_info: mir::SourceInfo) -> Option<Bx::DILocation> {
```

Comment:
One thing to point out: I need to turn a `mir::SourceInfo` into a `Bx::DILocation` and this was the only place I found that does that. I just marked the function public to also use it when generating the block (`compiler/rustc_codegen_ssa/src/mir/block.rs`), I hope that is fine.

Reply:
My guess is this should be fine (cc @compiler-errors).

Reply:
FYI, @compiler-errors, this is not being changed anymore with the new implementation.